086-monitoring-of-custom-resources.md

```diff
@@ -0,0 +1,125 @@
+During the initial phase when implementing this proposal, the support in the examples will be implemented at least for the existing resources for which Strimzi publishes already existing metric(s), additionally for `KafkaUser` and `KafkaTopic` custom resources.
+
+### Deployment
+
+Strimzi should provide guidance for existing fields in status object in order to indicate if a CR is having a problem or uses a deprecated configuration in the official documentation.
+By this users can extend monitoring of CRs on their own requirements.
+
+In general, an example ConfigMap should be provided in the examples (sub-)directory `metrics/kube-state-metrics` of [strimzi/strimzi-kafka-operator](https://github.com/strimzi/strimzi-kafka-operator/tree/main/examples/metrics/kube-state-metrics) to specify a good starting configuration for users.
+The ConfigMap should look like e.g.:
+
+```
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  labels:
+  name: strimzi-kube-state-metrics-config
+data:
+  config.yaml: |
+    spec:
+      resources:
+      - groupVersionKind:
+          group: kafka.strimzi.io
+          version: v1beta2
+          kind: KafkaTopic
+        metricNamePrefix: strimzi_kafka_topic
+        metrics:
+          - name: resource_info
+            help: "The current state of a Strimzi kafka topic resource"
+            each:
+              type: Info
+              info:
+                labelsFromPath:
+                  name: [ metadata, name ]
+            labelsFromPath:
+              exported_namespace: [ metadata, namespace ]
+              partitions: [ spec, partitions ]
+              replicas: [ spec, replicas ]
+              ready: [ status, conditions, "[type=Ready]", status ]
+              generation: [ status, observedGeneration ]
+              topicId: [ status, topicId ]
+              topicName: [ status, topicName ]
+      - ...
+```
+
+For users deploying Kubernetes manifests as plain YAML, the ConfigMap works fine when creating a dedicated deployment of KSM.
+
+This works also fine when implement the configuration inside [kube-prometheus-stack](https://github.com/prometheus-community/helm-charts/tree/main/charts/kube-prometheus-stack) Helm chart.
+
+For (all other) Helm users, this should be left to the user implementing the KSM based monitoring, as there are many different Helm charts but all of them will handle the same ConfigMap in the end.
+This could be explained in a Blog Post with a static version of the [prometheus-community kube-state-metrics Helm chart](https://github.com/prometheus-community/helm-charts/tree/main/charts/kube-state-metrics/) at least once.
+
+In addition, Strimzi will define example `PrometheusRules` for all CRs for having a problem or uses a deprecated configuration in a simple way, so that a cluster operator can have a look for an error message in the status object of the CR.
```

Comment:
```suggestion
In addition, Strimzi will define example `PrometheusRules` for all CRs having issues or using deprecated configuration in a simple way, so that the Cluster Operator look for an error message in the status object of the CR.
```